### PR TITLE
Extend Keystore interface

### PR DESCRIPTION
**Pull request by iamnotacake**
Add methods `GetClientIDSymmetricKey` and `GetZoneIDSymmetricKey` that should return only the latest key that will be used to encrypt data by client / in zone.

Implemented new methods for filesystem keystore v1 and v2 (but no special impl for poison keys).
This should cover Redis as well.

## Checklist

- [x] Change is covered by automated tests
- [x] The [coding guidelines] are followed
- [x] ~Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes~
- [x] ~CHANGELOG.md is updated (in case of notable or breaking changes)~
- [x] CHANGELOG_DEV.md is updated
- [x] ~Benchmark results are attached (if applicable)~
- [x] ~[Example projects and code samples] are up-to-date (in case of API changes)~

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: crypto/poison_detector.go [hunk 1]
```diff
@@ -78,22 +78,50 @@ func NewPoisonRecordKeyStoreWrapper(keyStore keystore.RecordProcessorKeyStore) k
...
-// GetClientIDSymmetricKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetClientIDSymmetricKeys implementation of keystore.PrivateKeyStore for poison records keys
 func (p PoisonRecordKeyStoreWrapper) GetClientIDSymmetricKeys([]byte) ([][]byte, error) {
 	return p.keyStore.GetPoisonSymmetricKeys()
 }
 
-// GetServerDecryptionPrivateKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetClientIDEncryptionKey implementation of keystore.PrivateKeyStore for poison records symmetric key
+func (p PoisonRecordKeyStoreWrapper) GetClientIDEncryptionKey([]byte) ([]byte, error) {
+	encryptionKeys, err := p.keyStore.GetPoisonSymmetricKeys()
```
Comment (Lagovas) [inline crypto/poison_detector.go:88]: will be great to fetch here 1 key too instead of array of keys

File: crypto/poison_detector.go [hunk 2]
```diff
@@ -78,22 +78,50 @@ func NewPoisonRecordKeyStoreWrapper(keyStore keystore.RecordProcessorKeyStore) k
...
-// GetZoneIDSymmetricKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetZoneIDSymmetricKeys implementation of keystore.PrivateKeyStore for poison records keys
 func (p PoisonRecordKeyStoreWrapper) GetZoneIDSymmetricKeys(id []byte) ([][]byte, error) {
 	return p.keyStore.GetPoisonSymmetricKeys()
 }
 
-// GetZonePrivateKeys implementation of keystore.PrivateKeyStore with for poison records keys
+// GetZoneIDEncryptionKey implementation of keystore.PrivateKeyStore for poison records symmetric key
+func (p PoisonRecordKeyStoreWrapper) GetZoneIDEncryptionKey(id []byte) ([]byte, error) {
+	encryptionKeys, err := p.keyStore.GetPoisonSymmetricKeys()
```
Comment (Lagovas) [inline crypto/poison_detector.go:112]: and here

File: keystore/filesystem/server_keystore.go [hunk 3]
```diff
@@ -975,18 +975,39 @@ func (store *KeyStore) getSymmetricKeys(id []byte, keyname string) ([][]byte, er
...
+
+// GetClientIDSymmetricKeys return symmetric keys for specified client id
 func (store *KeyStore) GetClientIDSymmetricKeys(id []byte) ([][]byte, error) {
 	keyName := getClientIDSymmetricKeyName(id)
 	return store.getSymmetricKeys(id, keyName)
 }
 
-// GetZoneIDSymmetricKeys return symmetric key for specified zone id
+// GetClientIDEncryptionKey return latest symmetric key for encryption by specified client id
+func (store *KeyStore) GetClientIDEncryptionKey(id []byte) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (iamnotacake) [reply keystore/filesystem/server_keystore.go:994]: Then, do the same with `GetZoneIDEncryptionKey -> GetZoneIDSymmetricKey`?

[the remark this answers is not in the document]
Comment (Lagovas) [reply keystore/filesystem/server_keystore.go:994]: sure

File: crypto/poison_detector.go (hunk 1 above)
Comment (iamnotacake) [reply crypto/poison_detector.go:88]: This one seems kinda confusing for me. For filesystem keystores v1/v2 the methods were inside corresponding packages, I was able to use specific functions to get the key (one to read from file or one to read it from keyring). But here we're in different package that only interact with interfaces. Maybe missing something :c

Comment (Lagovas) [reply crypto/poison_detector.go:88]: so we need extend keystore interface related to poison records too with similar function that returns 1 key

**Review by Lagovas — CHANGES_REQUESTED** [review]

**Review by Lagovas — APPROVED** [review]
merge after successful test run

**Pull request merged**